Sources/EduIDExpansion/Flows/CreateEduID/ViewModels/CreatePincodeAndBiometricAccessViewModel.swift

```diff
@@ -84,37 +78,47 @@ final class CreatePincodeAndBiometricAccessViewModel: NSObject {
         Task {
             do{
                 let enrolment = try await TiqrControllerAPI.startEnrollmentWithRequestBuilder()
-                    .addHeader(name: Constants.Headers.authorization, value: keychain.getString(for: Constants.KeyChain.accessToken))
+                    .addHeader(name: Constants.Headers.authorization, value: keychain.getString(for: Constants.KeyChain.accessToken) ?? "")
                     .execute()
                     .body
                 
                 ServiceContainer.sharedInstance().challengeService.startChallenge(fromScanResult: enrolment.url ?? "") { [weak self] type, object, error in
                     guard let self else { return }
+                    self.secondEnteredPin.removeLast(2)
                     ServiceContainer.sharedInstance().challengeService.complete(object as! EnrollmentChallenge, usingBiometricID: withBiometrics, withPIN: self.pinToString(pinArray: self.secondEnteredPin)) { success, error in
                         if success {
+                            self.enrollmentChallenge = (object as? EnrollmentChallenge)
                             completion(true)
                         } else {
                             completion(false)
                         }
                     }
                 }
             } catch let error as NSError {
-                print(error.localizedDescription)
+                assertionFailure(error.localizedDescription)
                 completion(false)
             }
         }
     }
 }
 
 extension CreatePincodeAndBiometricAccessViewModel {
-    
-    //TODO: Should enable biometrics on the challenge
     @objc func requestBiometricAccess() {
         guard let viewController = self.viewController else { return }
         biometricService.useOnDeviceBiometricFeature { [weak self] success, error in
             guard let self else { return }
             if success {
-                (viewController.biometricApprovaldelegate as? CreateEduIDViewControllerDelegate)?.createEduIDViewControllerShowNextScreen(viewController: viewController)
+                if let enrolment = self.enrollmentChallenge {
+                    if let managedObject = enrolment.identity.managedObjectContext {
+                        enrolment.identity.biometricIDEnabled = NSNumber(value: 1)
+                        do {
+                            try managedObject.save()
+                            (viewController.biometricApprovaldelegate as? CreateEduIDViewControllerDelegate)?.createEduIDViewControllerShowNextScreen(viewController: viewController)
```

Comment:
Please use protocols

Reply:
there may be scenarios where an existing user has a challange stored on their device, but they don't have an AccessToken, so if that's the case the app will open the browser to get an Access Token, I noticed the UI doesn't get updated when users are redirected from a browser to the app.

Reply:
can you explain it further?